akka-persistence/src/main/scala/akka/persistence/AtLeastOnceDelivery.scala

```diff
@@ -197,7 +196,7 @@ trait AtLeastOnceDelivery extends Eventsourced {
    * This method will throw [[AtLeastOnceDelivery.MaxUnconfirmedMessagesExceededException]]
    * if [[numberOfUnconfirmed]] is greater than or equal to [[#maxUnconfirmedMessages]].
    */
-  def deliver(destination: ActorPath, deliveryIdToMessage: Long ⇒ Any): Unit = {
+  def deliver(destination: ActorPath)(deliveryIdToMessage: Long ⇒ Any): Unit = {
```

Comment:
Without splitting it up in 2 parameter lists inference breaks and is unable to figure out that `{ id => }` is `Long`.